### DISCUSSION TITLE
Module not found: Error: Can't resolve './swc.android-arm64.node'

### DISCUSSION THREAD

**Discussion by adamwong246** [swc-project/swc#6565] [Q&A]
I get this error all the time and I was wondering if anyone could tell me what it means and how to resolve it?
```
WARNING in ./tests/ClassicalReact/node_modules/@swc/core/binding.js 178:44-79
Module not found: Error: Can't resolve './swc.linux-x64-gnu.node' in '/Users/adam/Code/testeranto.ts/tests/ClassicalReact/node_modules/@swc/core'
 @ ./tests/ClassicalReact/node_modules/@swc/core/index.js 62:88-108
 @ ./tests/ClassicalReact/node_modules/terser-webpack-plugin/dist/utils.js 571:14-34
 @ ./tests/ClassicalReact/node_modules/terser-webpack-plugin/dist/index.js 28:4-22
 @ ./tests/ClassicalReact/node_modules/webpack/lib/config/defaults.js 1164:25-57
 @ ./tests/ClassicalReact/node_modules/webpack/lib/index.js 337:10-66
 @ ./tests/ClassicalReact/webpack-puppeteer.testeranto.test.ts 2:0-30 13:21-28
 @ ./tests/ClassicalReact/ClassicalComponent.webpack-puppeteer.test.ts 2:0-67 4:33-49
 @ ./tests/index.ts 7:0-91 16:4-21
 ```
 
Similar errors are thrown for `@swc/core-linux-x64-musl`, `@swc/core-linux-x64-gnu`, `@swc/core-freebsd-x64` and several others.

**Comment by adamwong246** [comment]
Attempting to install those missing packages returns
```
error @swc/core-win32-arm64-msvc@1.3.21: The platform "darwin" is incompatible with this module.
error Found incompatible module.
```

**Comment by mephju** [comment]
This happens when you "create" a `package-lock.json` on one platform (eg darwin/mac during dev) but then you install deps based on that `package-lock.json` on *another* platform (eg linux for prod). 

The workaround that I found for this issue is installing the specific bindings package for my prod env.
I added an *optional dependency* to my `package.json`
```json
"optionalDependencies": {
    "@swc/core-linux-x64-musl": "^1.3.21"
 },
```

So when I install this in dev env it will not fail bc it's an optional dependency. And in my prod env it will install just fine.
